### DISCUSSION TITLE
Extension properties

### DISCUSSION THREAD

**Discussion by mrlife** [dotnet/runtime#60086] [General]
Sometimes it would be useful to have an extension property, such as adding a property to List<T> that gets/sets an array of the same type.

For example, Blazor `<InputSelect>` now supports the `multiple` attribute and binds to an array. I prefer working with lists, so it would be nice if I could continue to use lists but have an `asArray` property that can be used to get or set the list.

```c#
List<int> myList = new();
int[] myArray = { 1, 2, 3, 4, 5, 6 };

myList.asArray = myArray;
```

**Comment by svick** [comment]
I don't understand what you're actually asking for. Could you clarify it, ideally by adding an example and explaining why it would be useful?

**Reply by mrlife** [reply]
Sure, I updated the top post.

**Comment by HaloFour** [comment]
API suggestions should be posted to the [runtime](https://github.com/dotnet/runtime) repository.

**Comment by DaZombieKiller** [comment]
This is effectively a subset of extension everything, which to my knowledge is subsumed by https://github.com/dotnet/csharplang/issues/1711

**Comment by huoyaoyuan** [comment]
The JS object model is way different from .NET (and most strongly/statically typed systems). It's not able to store arbitrary properties in an object.

**Reply by mrlife** [reply]
It would would with C# but not the JS interop?

**Reply by huoyaoyuan** [reply]
If a JS object exposes support for C# `dynamic` keyword, it would be possible. Otherwise, no.
